### DISCUSSION TITLE
Protocols with just a docstring

### DISCUSSION THREAD

**Discussion by padrepitufo** [microsoft/pyright#7379] [Q&A] [answered]
Is it expected that a protocol with an ellipsis vs a protocol without an ellipsis (but a docstring) should behave differently. In PyRight [one fails](https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAK4MYAxmADYBQNpFAhgM6NQAqApozABRFgnkKASgBcVKFAAm7YLE4wA%2BvW6N2FYABoo9EVC4gtAI136hUALQA%2BKIbCUxEiQDoX4qG%2BmyY8hYZVrNbRMYAxtgkDMrGzsKB0cAIkSZYHZSGBZ6OS4oMBQbRPiqIA). In MyPy they both perform the same.

<img width="1463" alt="image" src="https://github.com/microsoft/pyright/assets/91497353/6604f35d-f755-4115-998a-ef3062a302c6">

**Comment by rchiodo** [comment, marked as answer]
You forgot the `...` in the second version.

This code shows no error:

```python
from typing import Protocol

class Test(Protocol):
    def test_a(self, a, b) -> bool:
        ...

    def test_b(self, a, b) -> bool:
        ''' doc string '''
        ...
```

**Reply by rchiodo** [reply]
Oh maybe that was your point. Yeah Pyright needs the ellipsis, otherwise it looks like an unfinished function definiton.

**Reply by erictraut** [reply]
Yes, this is by design. However, it's an under-specified portion of the typing spec. This is on my (long) list of items to specify in the typing spec so we can increase consistency between type checkers. In the meantime, I'm not inclined to change pyright's behavior because I think it's principled and (at least internally) consistent. I don't want to have to change it and then revert it later if and when the typing community decides on a precise specification.

**Reply by padrepitufo** [reply]
this makes total sense. I was struggling to find anywhere in the spec or in the community with a definitive answer on what is right. I am just so happy you've been so responsive.
